From 6bb7c5b18b7d51cea19780bda18ae50f5c755d11 Mon Sep 17 00:00:00 2001
From: Jiang Xin <zhiyou.jx@alibaba-inc.com>
Date: Mon, 15 Mar 2021 17:41:08 +0800
Subject: [PATCH 39/42] crypto: set default crypto algorithm to 65

Set GIT_CRYPTO_ALGORITHM_DEFAULT to GIT_CRYPTO_ALGORITHM_EASY_AES,
which use 12-byte pack header.

Signed-off-by: Jiang Xin <zhiyou.jx@alibaba-inc.com>
---
 crypto.h                        | 3 ++-
 t/t0940/once-0001-crypto.sh     | 2 +-
 t/t0940/once-0002-crypto-map.sh | 2 +-
 3 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/crypto.h b/crypto.h
index 63b418ae61..5336abc411 100644
--- a/crypto.h
+++ b/crypto.h
@@ -56,7 +56,8 @@ enum agit_crypto_algo {
 						     production */
 	GIT_CRYPTO_ALGORITHM_EASY_AES = 65
 };
-#define GIT_CRYPTO_ALGORITHM_DEFAULT GIT_CRYPTO_ALGORITHM_AES
+
+#define GIT_CRYPTO_ALGORITHM_DEFAULT GIT_CRYPTO_ALGORITHM_EASY_AES
 
 /*
  * The 6th and 7th bytes are preserved for later use.
diff --git a/t/t0940/once-0001-crypto.sh b/t/t0940/once-0001-crypto.sh
index 50cd3746d7..3d3df99b63 100644
--- a/t/t0940/once-0001-crypto.sh
+++ b/t/t0940/once-0001-crypto.sh
@@ -104,7 +104,7 @@ test_expect_success 'encrypt text file using default algorithm' '
 		test-tool agit-crypto -z \
 		--secret "{plain}secret-token" \
 		-i text-file -o text-file.z.def &&
-	test_cmp text-file.z.2 text-file.z.def
+	test_cmp text-file.z.65 text-file.z.def
 '
 
 test_expect_success 'will fail with too short secret' '
diff --git a/t/t0940/once-0002-crypto-map.sh b/t/t0940/once-0002-crypto-map.sh
index a06d8d0c0a..c36802aa92 100644
--- a/t/t0940/once-0002-crypto-map.sh
+++ b/t/t0940/once-0002-crypto-map.sh
@@ -104,7 +104,7 @@ test_expect_success 'encrypt text file using default algorithm (with --mmap)' '
 		test-tool agit-crypto --mmap -z \
 		--secret "{plain}secret-token" \
 		-i text-file -o text-file.z.def &&
-	test_cmp text-file.z.2 text-file.z.def
+	test_cmp text-file.z.65 text-file.z.def
 '
 
 test_expect_success 'will fail with too short secret (with --mmap)' '
-- 
2.34.1.51.g7e1f4e9345.agit.6.5.6

